### PR TITLE
update form validation - no replacement IDs for bags

### PR DESCRIPTION
**Pull request by JoeArmani**
Related: https://github.com/episphere/connect/issues/903

-Update form validation structure: no replacement tube labels allowed in Bag ID fields.
-Update error messaging.

### CODE REVIEW

**Review by amber-emmes — APPROVED** [review]

File: src/events.js [hunk 1]
```diff
@@ -1076,28 +1076,38 @@ const redirectSpecimenPage = async (accessionID1, accessionID3, selectedVisit, f
...
+const checkFormAndSave = async (participantData, biospecimenData, shouldNavigateToReview) => {
+    try {
+        const inputFields = Array.from(document.getElementsByClassName('input-barcode-id'));
+        const isFormDataValid = inputFields.every(input => validateFormInputField(input, biospecimenData, true));
+        
+        isFormDataValid ?
+            await collectionSubmission(participantData, biospecimenData, shouldNavigateToReview) :
+            showTimedNotifications({ title: 'Data Errors Exist!', body: 'Please correct data entry errors in red before saving.' });
+    } catch (error) {
+        console.error("Error saving collection: ", error);
```
Comment (anthonypetersen) [inline src/events.js:1106]: Brian had a good question regarding `console.error()` earlier today... Do we know yet if these will be visible in DataDog? Or do we need to add something similar to what was done in the PWA?

File: src/events.js [hunk 2]
```diff
@@ -1256,39 +1266,12 @@ export const addEventBiospecimenCollectionFormEditAll = () => {
...
-                    errorMessage(input.id, 'Invalid Collection ID.');
-                }
-                else if (input.required && tubes.length === 0) {
-                    errorMessage(input.id, 'Invalid Full Specimen ID.');
-                }
-                else if (input.required && (tubes[0].id !== tubeID && !additionalTubeIDRequirement.regExp.test(tubeID))) {
-                    errorMessage(input.id, 'Invalid Full Specimen ID.');
-                }
-            }
+            validateFormInputField(input, biospecimenData);
```
Comment (anthonypetersen) [inline src/events.js:1274]: good refactor here

**Review by anthonypetersen — APPROVED** [review]
looks good - just the one question / comment about Data Dog visibility

File: src/events.js (hunk 1 above)
Comment (JoeArmani) [reply src/events.js:1106]: The console.error logs and the custom logs in the PWA both go to DataDog but neither is visible in the main error reporting screen. That main screen only shows uncaught errors.

The path to view console.error and custom logs is:
UX Monitoring -> Sessions Explorer -> Sort descending by error count.
Then click into the individual session to see console.error and custom logs.

**Pull request merged**